presto-main/src/main/java/com/facebook/presto/server/security/LdapFilter.java

```
@@ -0,0 +1,225 @@
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import javax.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+import java.security.Principal;
+import java.util.Base64;
+import java.util.Hashtable;
+import java.util.List;
+import java.util.Objects;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static javax.naming.Context.INITIAL_CONTEXT_FACTORY;
+import static javax.naming.Context.PROVIDER_URL;
+import static javax.naming.Context.SECURITY_AUTHENTICATION;
+import static javax.naming.Context.SECURITY_CREDENTIALS;
+import static javax.naming.Context.SECURITY_PRINCIPAL;
+
+public abstract class LdapFilter
+        implements Filter, LdapBinder
+{
+    private static final Logger LOG = Logger.get(LdapFilter.class);
+    public static final String AUTHENTICATION_TYPE = "basic";
+    public static final String LDAP_CONTEXT_FACTORY = "com.sun.jndi.ldap.LdapCtxFactory";
+    private final String ldapUrl;
+
+    protected LdapFilter(LdapServerConfig config)
+    {
+        ldapUrl = requireNonNull(config.getLdapUrl(), "ldap.url is null");
+        DirContext context = null;
+        try {
+            context = authenticate(getBasicEnvironment());
+        }
+        catch (NamingException e) {
+            throw Throwables.propagate(e);
+        }
+        finally {
+            closeQuietly(context);
+        }
+    }
+
+    private void closeQuietly(DirContext context)
+    {
+        try {
+            if (context != null) {
+                context.close();
+            }
+        }
+        catch (NamingException ex) {
+            LOG.error("Exception during Context.close()", ex);
```

Actually, I'm not sure it's useful to log here at all -- the close failed, but nothing can be done, and there isn't anything useful to be learned be logging the exception. Let's just ignore the exception and inline this method.